packages/injectable-react/ogre-tools-injectable-react.d.ts

```diff
@@ -9,28 +9,35 @@ declare module '@ogre-tools/injectable-react' {
 
   export const DiContextProvider: React.Provider<DiContainerProviderProps>;
 
-  export function withInjectables<Dependencies, Props = {}>(
-    Component: React.ElementType<Dependencies & Props>,
-    options: {
-      getProps: (di: DiContainer, props: Props) => Props & Dependencies;
-    },
-  ): React.ForwardRefExoticComponent<
-    Props & React.RefAttributes<any> & { children?: React.ReactNode }
-  >;
-
-  export function withInjectables<Dependencies, Props = {}>(
-    Component: React.ElementType<Dependencies & Props>,
-    options: {
-      getProps: (
-        di: DiContainer,
-        props: Props,
-      ) => PromiseLike<Props & Dependencies>;
-
-      getPlaceholder: () => JSX.Element;
-    },
-  ): React.ForwardRefExoticComponent<
-    Props & React.RefAttributes<any> & { children?: React.ReactNode }
-  >;
+  export interface WithInjectablesSyncOptions<Dependencies extends object, Props extends object> {
+    getProps: (di: DiContainer, props: Props) => Props & Dependencies;
+  }
+
+  export interface WithInjectablesAsyncOptions<Dependencies extends object, Props extends object> {
+    getProps: (di: DiContainer, props: Props) => Promise<Props & Dependencies>;
+    getPlaceholder: () => JSX.Element;
+  }
+
+  export interface WithInjectables {
+    <Dependencies extends object, Props extends object = {}>(
+      Component: React.ElementType<Dependencies & Props>,
+      options: WithInjectablesSyncOptions<Dependencies, Props>,
+    ): React.FunctionComponent<Props>;
+    <Dependencies extends object, Props extends object, Ref extends object>(
```

Comment:
Line breaks between all multi-line constructs with siblings. Championing this also to prettier is appreciated ;)